driver/csiplugin/connectors/rest_v2.go

```diff
@@ -718,7 +721,33 @@ func (s *SpectrumRestV2) ListCSIIndependentFilesets(ctx context.Context, filesys
 		return nil, err
 	}
 
-	return getFilesetResponse.Filesets, nil
+	filesets := getFilesetResponse.Filesets
+
+	emptyPages := Pages{}
+	for getFilesetResponse.Paging != emptyPages {
+		lastID := getLastID(getFilesetResponse.Paging.Next)
+
+		getFilesetURL := url + "&lastId=" + lastID + "&" + filter
+		getFilesetResponse = GetFilesetResponse_v2{}
+		klog.V(6).Infof("getFilesetURL with lastId [%v] ", getFilesetURL)
+		err := s.doHTTP(ctx, getFilesetURL, "GET", &getFilesetResponse, nil)
+		if err != nil {
+			klog.Errorf("[%s] Error in list fileset request with lastId: %v", utils.GetLoggerId(ctx), err)
+			return nil, err
+		}
+		filesets = append(filesets, getFilesetResponse.Filesets...)
+	}
+	return filesets, nil
+}
+
+func getLastID(nextURL string) string {
+	//Sample next URL format:
+	//"/scalemgmt/v2/filesystems/fs2/filesets?lastId=4368&fields=filesetName&filter=config.isInodeSpaceOwner=true,config.comment=Fileset created by IBM Container Storage Interface driver"
+	splitURL1 := strings.Split(nextURL, "&")
+	splitURL2 := strings.Split(splitURL1[0], "lastId=")
```

Comment:
@amdabhad do confirm with gui team if the sequence of lastId and fields always remain in this sequence.

Reply:
Confirmed with GUI, the sequence remains the same